modules/json/test/src/smithy4s/json/JsonCodecApiTests.scala

```diff
@@ -125,4 +105,38 @@ class JsonCodecApiTests extends FunSuite {
     }
   }
 
+  test(
+    "schemas bijected to Option should be encoded when options are"
+  ) {
+    val schemaWithOption = Schema
+      .struct[Option[String]]
+      .apply(
+        Schema.string.option
+          .field[Option[String]]("a", identity)
+      )(identity)
+
+    case class OptionalLike[+A](underlying: Option[A])
+
+    val schemaWithOptionEquivalent = Schema
+      .struct[OptionalLike[String]]
+      .apply(
+        Schema.string.option
+          .biject(OptionalLike(_))(_.underlying)
+          .field[OptionalLike[String]]("a", identity)
+      )(identity)
+
+    def go[A](input: A, schema: Schema[A]) = {
+      val capi = Json.payloadCodecs.withJsoniterCodecCompiler(
+        Json.jsoniter.withFieldFilter(FieldFilter.SkipUnsetOptions)
+      )
+
+      capi.encoders.fromSchema(schema).encode(input)
+    }
+
+    assertEquals(
+      go(None, schemaWithOption),
+      go(OptionalLike(None), schemaWithOptionEquivalent)
+    )
```

Comment:
Let's make this assertion consistent with the rest, as in change it to explicitly show a JSON payload. Whilst we're at it, let's also change it to cover exhaustively the changes in this fix, for instance : 

``` 
case class Options(option: Option[A], optional: Optional[A], refinedOptional: Optional[A], recursiveOption: Fix[Option])  
```

Reply:
done and done